crates/llvm-context/src/polkavm/evm/ether_gas.rs

```diff
@@ -40,22 +40,26 @@ pub fn balance<'ctx, D>(
 where
     D: Dependency + Clone,
 {
-    let balance_pointer = context.build_alloca(context.word_type(), "balance_pointer");
-    let address_pointer = context.build_alloca(context.word_type(), "address_pointer");
-    context.build_store(address_pointer, address)?;
+    let address_type = context.integer_type(revive_common::BIT_LENGTH_ETH_ADDRESS);
+    let address_pointer = context.build_alloca_at_entry(address_type, "address_pointer");
+    let address_truncated =
+        context
+            .builder()
+            .build_int_truncate(address, address_type, "address_truncated")?;
+    let address_swapped = context.build_byte_swap(address_truncated.into())?;
+    context.build_store(address_pointer, address_swapped)?;
 
+    let balance_pointer = context.build_alloca(context.word_type(), "balance_pointer");
```

Comment:
The reason is documented [LLVM performance guide](https://llvm.org/docs/Frontend/PerformanceTips.html#use-of-allocas). However we might as well just use a scratch buffer for those. Also need to think about zero-ing them out as this would give as the default zero value, which is assumed a lot in EVM, for free. A context helper taking care of this seems the most sensibly solution. But this is something for another PR :) So I'll leave it up to you.